azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -0,0 +1,494 @@
+            spc_client_id=spc_client_id,
+        )
+
+        self._add_secrets_to_secret_sync(
+            secrets=secrets_to_add,
+            secret_sync=opcua_secret_sync,
+            resource_group=resource_group,
+            spc_name=OPCUA_SPC_NAME,
+            secret_sync_name=OPCUA_CLIENT_CERT_SECRET_SYNC_NAME,
+        )
+
+        # update opcua extension
+        return self._update_client_secret_to_extension(
+            subject_name=subject_name,
+            application_uri=application_uri,
+        )
+
+    def _validate_key_files(self, public_key_file: str, private_key_file: str):
+        # validate public key file end with .der
+        validate_file_extension(public_key_file, [".der"])
+        # validate private key file end with .pem
+        validate_file_extension(private_key_file, [".pem"])
+
+        # validate public key and private key has matching file name without extension
+        public_key_name = os.path.basename(public_key_file)
+        public_key_name = os.path.splitext(public_key_name)[0]
+        private_key_name = os.path.basename(private_key_file)
+        private_key_name = os.path.splitext(private_key_name)[0]
+
+        if public_key_name != private_key_name:
+            raise ValueError(f"Public key file {public_key_name} and private key file {private_key_name} must match.")
+
+    def _process_fortos_yaml(self, object_text: str, secret_entry: Optional[dict] = None) -> str:
+        if object_text:
+            objects_obj = yaml.safe_load(object_text)
+        else:
+            objects_obj = {"array": []}
+        entry_text = yaml.safe_dump(secret_entry, indent=6)
+        objects_obj["array"].append(entry_text)
+        object_text = yaml.safe_dump(objects_obj, indent=6)
+        # TODO: formatting will be removed once fortos service fixes the formatting issue
+        return object_text.replace("\n- |", "\n    - |")
+
+    def _get_cl_resources(self, instance_name: str, resource_group: str) -> dict:
+        self.instance = self.instances.show(name=instance_name, resource_group_name=resource_group)
+        self.resource_map = self.instances.get_resource_map(self.instance)
+        custom_location = self.resource_client.resources.get_by_id(
+            resource_id=self.instance["extendedLocation"]["name"], api_version=CUSTOM_LOCATIONS_API_VERSION
+        )
+        cl_resources = self.resource_map.connected_cluster.get_aio_resources(custom_location_id=custom_location["id"])
+        return cl_resources
+
+    def _find_existing_spc(self, instance_name: str, cl_resources: List[dict]) -> dict:
+        secretsync_spc = None
+
+        if cl_resources:
+            for resource in cl_resources:
+                if resource["type"].lower() == "microsoft.secretsynccontroller/azurekeyvaultsecretproviderclasses":
+                    resource_id_container = parse_resource_id(resource["id"])
+                    secretsync_spc = self.ssc_mgmt_client.azure_key_vault_secret_provider_classes.get(
```

Comment:
why not return here? you dont do anything else like check if there is more than one, etc

Reply:
I do none check and raise below, otherwise it will be 3 repetitive code block for none check and raise in each add